net/peers/peer.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package peers contains helpers for looking up peers on a Tailscale network.
+package peers
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"os/exec"
+)
+
+// A Peer is another machine connected on a users network.
+type Peer struct {
+	Hostname string `json:"Hostname"`
+	TailAddr net.IP `json:"TailAddr"`
+}
+
+// A Queryer can be used to query for peers.
+type Queryer interface {
+	Query() ([]Peer, error)
+}
+
+// NewQueryer creates a new queryer object.
+func NewQueryer() Queryer {
```

Comment:
This style of defining an interface just to return it isn't really idiomatic Go. Generally you define an interface for something you accept.

This is a lot of indirection for something that could just be `func GetPeers() ([]Peer, error)`, no?

Or if there are plans for more extensibility, using a concrete type would give you places to add optional knobs in the future (so returning a `*Queryer` instead, at which point you can drop the constructor and just document that the Querier zero value is valid).